pkg/assume/assume.go

```diff
@@ -505,12 +507,20 @@ func AssumeCommand(c *cli.Context) error {
 			}
 
 			if retry {
-				for i := 0; i < 5; i++ {
+
+				b := sethRetry.NewFibonacci(time.Second)
+				b = sethRetry.WithMaxDuration(time.Minute*1, b)
+				err = sethRetry.Do(c.Context, b, func(ctx context.Context) (err error) {
 					creds, err = profile.AssumeTerminal(c.Context, configOpts)
 					if err == nil {
-						break
+						return sethRetry.RetryableError(err)
```

Comment:
I won't block merge on this but a couple of extra thoughts @meyerjrr:

- retrying on all errors could cause `assume` to feel as though it's hanging in the event of a network error, we could improve this in future by checking the error type here
- it would be good to print to the user something like `trying to assume the role... [time elapsed = 10s]` similar to Terraform so that they know we are retrying